src/app/shared/select/select-constraint-item/select-constraint-items-formatter.ts

```diff
@@ -0,0 +1,172 @@
+  [DateReadableFormatType.Yearly]: 'YYYY',
+  [DateReadableFormatType.Quarterly]: '[Q]Q YYYY',
+  [DateReadableFormatType.Weekly]: '[W]W YYYY',
+  [DateReadableFormatType.MonthYear]: 'MM.YYYY',
+  [DateReadableFormatType.Month]: 'MM',
+  [DateReadableFormatType.DayMonth]: 'DD.MM',
+  [DateReadableFormatType.Day]: 'DD',
+  [DateReadableFormatType.DayMonthYear]: 'DD.MM.YYYY',
+  [DateReadableFormatType.Hour]: 'HH',
+  [DateReadableFormatType.HourMinutes]: 'HH:mm',
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SelectConstraintItemsFormatter {
+  private readonly defaultTitle: string;
+
+  constructor(private i18n: I18n) {
+    this.defaultTitle = i18n({id: 'default', value: 'Default'});
+  }
+
+  public createItems(attribute: Attribute): SelectItemModel[] {
+    if (!attribute || !attribute.constraint) {
+      return [];
+    }
+
+    switch (attribute.constraint.type) {
+      case ConstraintType.DateTime:
+        return this.createDateConstraintItems(attribute.constraint.config as DateTimeConstraintConfig);
+      default:
+        return [];
+    }
+  }
+
+  private createDateConstraintItems(config: DateTimeConstraintConfig): SelectItemModel[] {
+    const formatTypes = createDateConstraintOverrideFormatTypes(config);
+    const defaultItem: SelectItemModel = {id: null, value: this.defaultTitle};
+    return [
+      defaultItem,
+      ...formatTypes
+        .map(type => ({type, format: dateReadableFormatsMap[type]}))
+        .filter(({format}) => format !== config.format)
+        .map(({type, format}) => ({id: {format}, value: this.translateDateReadableFormatType(type)})),
+    ];
+  }
+
+  public checkValidConstraintOverride(
+    constraint: Constraint,
+    overrideConfig: Partial<ConstraintConfig>
+  ): Partial<ConstraintConfig> {
+    switch (constraint.type) {
+      case ConstraintType.DateTime:
+        const configFormat =
+          overrideConfig &&
+          <DateTimeConstraintConfig>overrideConfig &&
+          (<DateTimeConstraintConfig>overrideConfig).format;
+        if (configFormat) {
+          const validFormats = createDateConstraintOverrideFormatTypes(
+            constraint.config as DateTimeConstraintConfig
+          ).map(type => dateReadableFormatsMap[type]);
+          const overrideFormat = validFormats.includes(configFormat)
+            ? this.translateDateReadableFormat(configFormat)
+            : null;
+          return overrideFormat && {format: overrideFormat};
+        }
+        return null;
+      default:
+        return false;
+    }
+  }
+
+  private translateDateReadableFormat(format: string): string {
+    const formatType = Object.entries(DateReadableFormatType).find(([, value]) => value === format);
+    if (formatType && formatType[0] === DateReadableFormatType.Weekly) {
+      return this.i18n(
+        {
+          id: 'select.constraint.items.date.format',
+          value: '{format, select, weekly {[W]W YYYY} other {} }',
```

Comment:
Is this the only format that needs translation? Should not we translate all of them?

Reply:
Yes this is the only one for now.